### DISCUSSION TITLE
Unable to delete applets or packages from cards

### DISCUSSION THREAD

**Discussion by Laura-IsANoun** [martinpaljak/GlobalPlatformPro#324] [Q&A]
I'm struggling to delete apps from an ACOSJ card.

I use the gp --delete XX XX XX XX XX command

then get this message:
'Could not delete AID: XX XX XX XX XX
DELETE failed: 0x6F00'

Why could this be? I'm running the latest version of Global Platform Pro

**Comment by martinpaljak** [comment]
As always, do provide a log as instructed (also the contents of the card before deletion).

This (seems like/is) a generic exception from the card, so hard to say why.

**Reply by Laura-IsANoun** [reply]
Contents of card before deletion and log:
![card contents](https://github.com/martinpaljak/GlobalPlatformPro/assets/35061240/432cf2ee-6544-4e4e-b5b9-3818ea55f30c)
![log](https://github.com/martinpaljak/GlobalPlatformPro/assets/35061240/b5798adf-8dd7-4b85-883f-1208610f1c92)

**Comment by martinpaljak** [comment]
Please paste text not screenshots.

What if you try to delete the package, not the applet? that is `gp -delete 0703199600 -f -dv`

**Reply by Laura-IsANoun** [reply]
Sorry!

If I try to delete the package I get this:
```GlobalPlatformPro` v20.01.23-0-g5ad373b
Running on Windows 7 6.1 amd64, Java 1.8.0_191 by Oracle Corporation
# Detected readers from JNA2PCSC
[*] Alcor Micro USB Smart Card Reader 0
[ ] JAVACOS Virtual Contact Reader 0
[ ] JAVACOS Virtual Contactless Reader 1
SCardConnect("Alcor Micro USB Smart Card Reader 0", T=*) -> T=0, 3B690002414
34A76313031
SCardBeginTransaction("Alcor Micro USB Smart Card Reader 0")
Reader: Alcor Micro USB Smart Card Reader 0
ATR: 3B69000241434F534A76313031
More information about your card:
http://smartcard-atr.appspot.com/parse?ATR=3B69000241434F534A76313031
A>> T=0 (4+0000) 00A40400 00
A<< (0094+2) (172ms) 6F5C8408A000000151000000A550734A06072A864886FC6B01600C0
A864886FC6B02020201630906072A864886FC6B03640B06092A864886FC6B040255650B06092
886FC6B020103660C060A2B060104012A026E01039F6501FF 9000
[TRACE] GPSession - [6F]
[TRACE] GPSession - [84] A000000151000000
[TRACE] GPSession - [A5]
[TRACE] GPSession - [73]
[TRACE] GPSession - [06] 2A864886FC6B01
[TRACE] GPSession - [60]
[TRACE] GPSession - [06] 2A864886FC6B02020201
[TRACE] GPSession - [63]
[TRACE] GPSession - [06] 2A864886FC6B03
[TRACE] GPSession - [64]
[TRACE] GPSession - [06] 2A864886FC6B040255
[TRACE] GPSession - [65]
[TRACE] GPSession - [06] 2A864886FC6B020103
[TRACE] GPSession - [66]
[TRACE] GPSession - [06] 2B060104012A026E0103
[TRACE] GPSession - [9F65] FF
[DEBUG] GPSession - Auto-detected ISD: A000000151000000
Warning: no keys given, using default test key 404142434445464748494A4B4C4D4
[WARN] PlaintextKeys - Don't know how to calculate KCV, defaulting to SCP02
[WARN] PlaintextKeys - Don't know how to calculate KCV, defaulting to SCP02
[WARN] PlaintextKeys - Don't know how to calculate KCV, defaulting to SCP02
[INFO] GPSession - Using card master keys: ENC=404142434445464748494A4B4C4D4
(KCV: 8BAF47) MAC=404142434445464748494A4B4C4D4E4F (KCV: 8BAF47) DEK=4041424
5464748494A4B4C4D4E4F (KCV: 8BAF47) for null
[TRACE] GPSession - Generated host challenge: 71076D5ADCA4CE37
A>> T=0 (4+0008) 80500000 08 71076D5ADCA4CE37 00
A<< (0028+2) (109ms) 0000026501830395366220020147699B18CBD7EA5EDB5C3D7ABA3E1
00
[DEBUG] GPSession - Host challenge: 71076D5ADCA4CE37
[DEBUG] GPSession - Card challenge: 0147699B18CBD7EA
[DEBUG] GPSession - Card reports SCP02 with key version 32 (0x20)
[INFO] GPSession - Diversified card keys: ENC=404142434445464748494A4B4C4D4E
KCV: 8BAF47) MAC=404142434445464748494A4B4C4D4E4F (KCV: 8BAF47) DEK=40414243
464748494A4B4C4D4E4F (KCV: 8BAF47) for SCP02
[INFO] GPSession - Session keys: ENC=858B4C73B1B64568D94474F0308D91D5 MAC=FE
FCCB99D94186A4784398990BEF RMAC=F25F8CEBA468CC90C8E9B4BDE7896AF2, card keys=
404142434445464748494A4B4C4D4E4F (KCV: 8BAF47) MAC=404142434445464748494A4B4
E4F (KCV: 8BAF47) DEK=404142434445464748494A4B4C4D4E4F (KCV: 8BAF47) for SCP
[DEBUG] GPSession - Verified card cryptogram: 5EDB5C3D7ABA3E1E
[DEBUG] GPSession - Calculated host cryptogram: 6A6D0AFEDC60A995
[TRACE] SCP02Wrapper - MAC input: 84820100106A6D0AFEDC60A995
A>> T=0 (4+0016) 84820100 10 6A6D0AFEDC60A995BDF3EEEE3D7A58C5
A<< (0000+2) (62ms) 9000
[TRACE] SCP02Wrapper - MAC input: 84F280020A4F00
A>> T=0 (4+0010) 84F28002 0A 4F002E8B2C90C0D2F42B 00
A<< (0021+2) (93ms) E3134F08A0000001510000009F700101C5039EDE00 9000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] A000000151000000
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [C5] 9EDE00
[TRACE] SCP02Wrapper - MAC input: 84F240020A4F00
A>> T=0 (4+0010) 84F24002 0A 4F0096EB1B0926DB4427 00
A<< (0042+2) (140ms) E3114F060703199600009F700107C503000000E3154F0A4A43416C6
57374319F700107C503000000 9000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 070319960000
[TRACE] GPRegistry - [9F70] 07
[TRACE] GPRegistry - [C5] 000000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 4A43416C675465737431
[TRACE] GPRegistry - [9F70] 07
[TRACE] GPRegistry - [C5] 000000
[TRACE] SCP02Wrapper - MAC input: 84F210020A4F00
A>> T=0 (4+0010) 84F21002 0A 4F0016DDAA9530806EF0 00
A<< (0142+2) (252ms) E3174F07A00000015153509F7001018408A000000151535041E3214
1636F736A2E636F739F700101841061636F736A2E636F732E434F53457874E31E4F06A000000
19F7001018407A000000333A1018407A000000333A102E3134F0507031996009F70010184060
9960000E31B4F094A43416C67546573749F700101840A4A43416C675465737431 9000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] A0000001515350
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [84] A000000151535041
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 61636F736A2E636F73
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [84] 61636F736A2E636F732E434F53457874
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] A000000333A1
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [84] A000000333A101
[TRACE] GPRegistry - [84] A000000333A102
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 0703199600
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [84] 070319960000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 4A43416C6754657374
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [84] 4A43416C675465737431
[TRACE] SCP02Wrapper - MAC input: 84F220020A4F00
A>> T=0 (4+0010) 84F22002 0A 4F00AB1EA1CA3352814D 00
A<< (0076+2) (172ms) E30D4F07A00000015153509F700101E30F4F0961636F736A2E636F7
00101E30C4F06A000000333A19F700101E30B4F0507031996009F700101E30F4F094A43416C6
573749F700101 9000
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] A0000001515350
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 61636F736A2E636F73
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] A000000333A1
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 0703199600
[TRACE] GPRegistry - [9F70] 01
[TRACE] GPRegistry - [E3]
[TRACE] GPRegistry - [4F] 4A43416C6754657374
[TRACE] GPRegistry - [9F70] 01
[DEBUG] GPRegistry - Registry already contains PKG: A0000001515350, 1
[DEBUG] GPRegistry - Registry already contains PKG: 61636F736A2E636F73, 1
[DEBUG] GPRegistry - Registry already contains PKG: A000000333A1, 1
[DEBUG] GPRegistry - Registry already contains PKG: 0703199600, 1
[DEBUG] GPRegistry - Registry already contains PKG: 4A43416C6754657374, 1
[TRACE] DMTokenGenerator - No private key for token generation provided
[TRACE] GPSession - TLV payload:
[TRACE] GPSession - [4F] 0703199600
[TRACE] SCP02Wrapper - MAC input: 84E400800F4F050703199600
A>> T=0 (4+0015) 84E40080 0F 4F050703199600901B33AE1B94A803
A<< (0000+2) (5s908ms) 6F00
Could not delete AID: 0703199600
SCardEndTransaction("Alcor Micro USB Smart Card Reader 0")
SCardDisconnect("Alcor Micro USB Smart Card Reader 0", true) tx:124/rx:419```
`

**Reply by martinpaljak** [reply]
You need to ask your card vendor why the delete is failing. A buggy card maybe?

**Reply by Laura-IsANoun** [reply]
I will do - thank you. It did work before so I am wondering if perhaps it's something within my apps code
